astroplan/plots/time_dependent.py

```diff
@@ -176,6 +183,23 @@ def plot_airmass(targets, observer, time, ax=None, style_kwargs=None,
 
     if not isinstance(targets, Sequence):
         targets = [targets]
+    
+    # Firstly, check that the constraints are valid if any.
+    if constraints is not None:
+        if isinstance(constraints, Constraint):
+            constraints = [constraints]
+    
+        assert isinstance(constraints, (list, tuple, np.ndarray)), "constraints must be an array-like object or a single Constraint instance."
+        for cons in constraints:
+            assert isinstance(cons, Constraint), "All constraint objects must be Constraints"
+
+            # Make sure they're boolean
+            assert hasattr(cons, 'boolean_constraint'), "Each constraints element must have the 'boolean_constraint attribute."
+            assert cons.boolean_constraint, "Cannot work with non-boolean constraints."
+            # Then make sure they're not some weird user-generated constraint. 
+            assert hasattr(cons, 'compute_constraint'), "Constraints must have a 'compute_constraint' method."
+            
+
```

Comment:
Rather than including `assert`s in the function here, a more common approach is to extract the elements of the list that meet the validation criteria, without raising an error.